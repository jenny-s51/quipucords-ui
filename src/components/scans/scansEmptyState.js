import React from 'react';
import PropTypes from 'prop-types';
import {
  Button,
  EmptyState,
  EmptyStateBody,
  EmptyStateIcon,
  EmptyStatePrimary,
  EmptyStateVariant,
  Title
} from '@patternfly/react-core';
import { AddCircleOIcon } from '@patternfly/react-icons';
import { Grid, Row } from 'patternfly-react';
import SourcesEmptyState from '../sources/sourcesEmptyState';
import helpers from '../../common/helpers';
import { connectRouter, reduxActions, reduxSelectors, reduxTypes, store } from '../../redux';
import { translate } from '../i18n/i18n';

/**
 * Return a scans empty state.
 */
class ScansEmptyState extends React.Component {
  componentDidMount() {
    const { getScansSources } = this.props;

    getScansSources();
  }

  onAddSource = () => {
    const { history, sourcesExist } = this.props;

    if (sourcesExist) {
      history.push('/sources');
    } else {
      store.dispatch({
        type: reduxTypes.sources.CREATE_SOURCE_SHOW
      });
    }
  };

  render() {
    const { sourcesExist, t } = this.props;

    if (sourcesExist) {
      return (
        <Grid fluid>
          <Row>
            <EmptyState className="quipucords-empty-state" variant={EmptyStateVariant.large}>
              <EmptyStateIcon icon={AddCircleOIcon} />
              <Title headingLevel="h1">{t('view.empty-state', { context: ['title', 'scans'] })}</Title>
              <EmptyStateBody>{t('view.empty-state', { context: ['description', 'scans'] })}</EmptyStateBody>
              <EmptyStatePrimary>
                <Button onClick={this.onAddSource}>
                  {t('view.empty-state', { context: ['label', 'source-navigate'] })}
                </Button>
              </EmptyStatePrimary>
            </EmptyState>
          </Row>
        </Grid>
      );
    }

    return <SourcesEmptyState onAddSource={this.onAddSource} />;
  }
}

/**
 * Prop types
 *
 * @type {{getScansSources: Function, sourcesExist: boolean, t: Function, history: object}}
 */
ScansEmptyState.propTypes = {
  getScansSources: PropTypes.func,
  history: PropTypes.shape({
    push: PropTypes.func
  }),
  sourcesExist: PropTypes.bool,
  t: PropTypes.func
};

/**
 * Default props
 *
 * @type {{getScansSources: Function, sourcesExist: boolean, t: translate, history: object}}
 */
ScansEmptyState.defaultProps = {
  getScansSources: helpers.noop,
  history: {},
  sourcesExist: false,
  t: translate
};

const mapDispatchToProps = dispatch => ({
  getScansSources: queryObj => dispatch(reduxActions.sources.getScansSources(queryObj))
});

const makeMapStateToProps = () => {
  const scansEmptyState = reduxSelectors.scans.makeScansEmptyState();

  return (state, props) => ({
    ...scansEmptyState(state, props)
  });
};

const ConnectedScansEmptyState = connectRouter(makeMapStateToProps, mapDispatchToProps)(ScansEmptyState);

export { ConnectedScansEmptyState as default, ConnectedScansEmptyState, ScansEmptyState };
