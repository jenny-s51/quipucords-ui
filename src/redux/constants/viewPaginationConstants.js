const VIEW_PAGE = 'VIEW_PAGE';
const SET_PER_PAGE = 'SET_PER_PAGE';

const viewPaginationTypes = {
  SET_PER_PAGE,
  VIEW_PAGE
};

export { viewPaginationTypes as default, viewPaginationTypes, SET_PER_PAGE, VIEW_PAGE };
