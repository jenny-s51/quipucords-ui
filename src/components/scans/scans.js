import React from 'react';
import PropTypes from 'prop-types';
import { Alert, AlertVariant, Button, ButtonVariant, EmptyState, Spinner } from '@patternfly/react-core';
import { IconSize } from '@patternfly/react-icons';
import { Modal, ModalVariant } from '../modal/modal';
import { reduxTypes, storeHooks } from '../../redux';
import ViewToolbar from '../viewToolbar/viewToolbar';
import ViewPaginationRow from '../viewPaginationRow/viewPaginationRow';
import { ScansEmptyState } from './scansEmptyState';
import { ScanFilterFields, ScanSortFields } from './scanConstants';
import { translate } from '../i18n/i18n';
import { Table } from '../table/table';
import { scansTableCells } from './scansTableCells';
import { useGetScans, useOnExpand, useOnRefresh, useOnScanAction, useOnSelect } from './scansContext';
import { Tooltip } from '../tooltip/tooltip';

const VIEW_ID = 'scans';

// ToDo: review onMergeReports, renderToolbarActions being standalone with upcoming toolbar updates
// ToDo: review items being selected and the page polling. Randomized dev data gives the appearance of an issue. Also applies to sources selected items
/**
 * A scans view.
 *
 * @param {object} props
 * @param {Function} props.t
 * @param {Function} props.useGetScans
 * @param {Function} props.useOnExpand
 * @param {Function} props.useOnRefresh
 * @param {Function} props.useOnScanAction
 * @param {Function} props.useOnSelect
 * @param {Function} props.useDispatch
 * @param {Function} props.useSelectors
 * @param {string} props.viewId
 * @returns {React.ReactNode}
 */
const Scans = ({
  t,
  useGetScans: useAliasGetScans,
  useOnExpand: useAliasOnExpand,
  useOnRefresh: useAliasOnRefresh,
  useOnScanAction: useAliasOnScanAction,
  useOnSelect: useAliasOnSelect,
  useDispatch: useAliasDispatch,
  useSelectors: useAliasSelectors,
  viewId
}) => {
  const dispatch = useAliasDispatch();
  const onExpand = useAliasOnExpand();
  const onRefresh = useAliasOnRefresh();
  const { onCancel, onDownload, onPause, onRestart, onStart } = useAliasOnScanAction();
  const onSelect = useAliasOnSelect();
  const { pending, error, errorMessage, date, data, selectedRows = {}, expandedRows = {} } = useAliasGetScans();
  const [viewOptions = {}] = useAliasSelectors([
    ({ viewOptions: stateViewOptions }) => stateViewOptions[reduxTypes.view.SCANS_VIEW]
  ]);
  const isActive = viewOptions?.activeFilters?.length > 0 || data?.length > 0 || false;

  /**
   * Toolbar actions onScanSources
   *
   * @event onMergeReports
   */
  const onMergeReports = () => {
    dispatch({
      type: reduxTypes.scans.MERGE_SCAN_DIALOG_SHOW,
      show: true,
      scans: Object.values(selectedRows).filter(val => val !== null)
    });
  };

  /**
   * Return toolbar actions.
   *
   * @returns {React.ReactNode}
   */
  const renderToolbarActions = () => (
    <Tooltip content={t('table.tooltip', { context: ['merge-reports'] })}>
      <Button
        variant={ButtonVariant.primary}
        isDisabled={Object.values(selectedRows).filter(val => val !== null).length <= 1}
        onClick={onMergeReports}
      >
        {t('table.label', { context: ['merge-reports'] })}
      </Button>
    </Tooltip>
  );

  if (pending) {
    return (
      <Modal variant={ModalVariant.medium} backdrop={false} isOpen disableFocusTrap>
        <Spinner isSVG size={IconSize.lg} />
        <div className="text-center">{t('view.loading', { context: viewId })}</div>
      </Modal>
    );
  }

  if (error) {
    return (
      <EmptyState className="quipucords-empty-state__alert">
        <Alert variant={AlertVariant.danger} title={t('view.error', { context: viewId })}>
          {t('view.error-message', { context: [viewId], message: errorMessage })}
        </Alert>
      </EmptyState>
    );
  }

  return (
    <div className="quipucords-view-container">
      {isActive && (
        <React.Fragment>
          <ViewToolbar
            viewType={reduxTypes.view.SCANS_VIEW}
            filterFields={ScanFilterFields}
            sortFields={ScanSortFields}
            onRefresh={() => onRefresh()}
            lastRefresh={new Date(date).getTime()}
            actions={renderToolbarActions()}
            itemsType="Scan"
            itemsTypePlural="Scans"
            selectedCount={viewOptions.selectedItems?.length}
            {...viewOptions}
          />
          <ViewPaginationRow viewType={reduxTypes.view.SCANS_VIEW} {...viewOptions} />
        </React.Fragment>
      )}
      <div className="quipucords-list-container">
        <Table
          onExpand={onExpand}
          onSelect={onSelect}
          rows={data?.map((item, index) => ({
            isSelected: (selectedRows?.[item.id] && true) || false,
            item,
            cells: [
              {
                content: scansTableCells.description(item),
                dataLabel: t('table.header', { context: ['description'] })
              },
              {
                content: scansTableCells.scanStatus(item, { viewId }),
                width: 20,
                dataLabel: t('table.header', { context: ['scan'] })
              },
              {
                ...scansTableCells.okHostsCellContent(item, { viewId }),
                isExpanded: expandedRows?.[item.id] === 2,
                width: 8,
                dataLabel: t('table.header', { context: ['success', viewId] })
              },
              {
                ...scansTableCells.failedHostsCellContent(item, { viewId }),
                isExpanded: expandedRows?.[item.id] === 3,
                width: 8,
                dataLabel: t('table.header', { context: ['failed', viewId] })
              },
              {
                ...scansTableCells.sourcesCellContent(item, { viewId }),
                isExpanded: expandedRows?.[item.id] === 4,
                width: 8,
                dataLabel: t('table.header', { context: ['sources'] })
              },
              {
                ...scansTableCells.scansCellContent(item, { viewId }),
                isExpanded: expandedRows?.[item.id] === 5,
                width: 8,
                dataLabel: t('table.header', { context: ['scan-jobs'] })
              },
              {
                style: { textAlign: 'right' },
                content: scansTableCells.actionsCell({
                  isFirst: index === 0,
                  isLast: index === data.length - 1,
                  item,
                  onCancel: () => onCancel(item),
                  onDownload: () => onDownload(item),
                  onRestart: () => onRestart(item),
                  onPause: () => onPause(item),
                  onStart: () => onStart(item)
                }),
                isActionCell: true
              }
            ]
          }))}
        >
          <ScansEmptyState viewId={viewId} />
        </Table>
      </div>
    </div>
  );
};

/**
 * Prop types
 *
 * @type {{useOnEdit: Function, useOnSelect: Function, viewId: string, t: Function, useOnRefresh: Function, useOnScan: Function,
 *     useDispatch: Function, useOnDelete: Function, useOnExpand: Function, useGetSources: Function, useSelectors: Function,
 *     useOnShowAddSourceWizard: Function}}
 */
Scans.propTypes = {
  t: PropTypes.func,
  useDispatch: PropTypes.func,
  useGetScans: PropTypes.func,
  useOnExpand: PropTypes.func,
  useOnRefresh: PropTypes.func,
  useOnScanAction: PropTypes.func,
  useOnSelect: PropTypes.func,
  useSelectors: PropTypes.func,
  viewId: PropTypes.string
};

/**
 * Default props
 *
 * @type {{useOnEdit: Function, useOnSelect: Function, viewId: string, t: translate, useOnRefresh: Function, useOnScan: Function,
 *     useDispatch: Function, useOnDelete: Function, useOnExpand: Function, useGetSources: Function, useSelectors: Function,
 *     useOnShowAddSourceWizard: Function}}
 */
Scans.defaultProps = {
  t: translate,
  useDispatch: storeHooks.reactRedux.useDispatch,
  useGetScans,
  useOnExpand,
  useOnRefresh,
  useOnScanAction,
  useOnSelect,
  useSelectors: storeHooks.reactRedux.useSelectors,
  viewId: VIEW_ID
};

export { Scans as default, Scans, VIEW_ID };
