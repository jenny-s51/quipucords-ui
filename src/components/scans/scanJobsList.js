import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import { Alert, AlertVariant, EmptyState, EmptyStateVariant, Spinner } from '@patternfly/react-core';
import { Grid, Icon } from 'patternfly-react';
import { connect, reduxActions, reduxSelectors } from '../../redux';
import { helpers } from '../../common/helpers';
import { dictionary } from '../../constants/dictionaryConstants';
import { apiTypes } from '../../constants/apiConstants';
import ScanDownload from './scanDownload';
import { translate } from '../i18n/i18n';

/**
 * Return a scan jobs listing for "jobs".
 */
class ScanJobsList extends React.Component {
  state = {
    currentPage: 1,
    queryObject: {
      [apiTypes.API_QUERY_PAGE]: 1,
      [apiTypes.API_QUERY_PAGE_SIZE]: 100,
      [apiTypes.API_QUERY_ORDERING]: '-end_time'
    }
  };

  componentDidMount() {
    const { queryObject } = this.state;
    const { getScanJobs, id } = this.props;

    getScanJobs(id, queryObject);
  }

  onScrollList = event => {
    const { target } = event;
    const { currentPage, queryObject } = this.state;
    const { getScanJobs, id, isMoreResults, pending } = this.props;

    const bottom = target.scrollHeight - target.scrollTop === target.clientHeight;

    if (bottom && !pending && isMoreResults) {
      const newPage = currentPage + 1;

      const updatedQueryObject = {
        [apiTypes.API_QUERY_PAGE]: newPage
      };

      this.setState(
        {
          currentPage: newPage
        },
        () => {
          getScanJobs(id, { ...queryObject, ...updatedQueryObject });
        }
      );
    }
  };

  render() {
    const { error, errorMessage, mostRecentId, pending, scanJobsList, t } = this.props;

    if (pending) {
      return (
        <EmptyState className="quipucords-empty-state" variant={EmptyStateVariant.large}>
          <Spinner isSVG size="sm" /> {t('view.loading')}
        </EmptyState>
      );
    }

    if (error) {
      return (
        <EmptyState className="quipucords-empty-state__alert">
          <Alert isInline isPlain variant={AlertVariant.danger} title={t('view.error', { context: 'scan-jobs' })}>
            {t('view.error-message', { context: ['scan-jobs'], message: errorMessage })}
          </Alert>
        </EmptyState>
      );
    }

    const iconProps = status => {
      const { type, name, classNames } = helpers.scanStatusIcon(status);
      return {
        type,
        name,
        className: cx('scan-job-status-icon', classNames)
      };
    };

    return (
      <div className="quipucords-infinite-results">
        <Grid fluid onScroll={this.onScrollList} className="quipucords-infinite-list">
          {scanJobsList.map(
            item =>
              mostRecentId !== item.id && (
                <Grid.Row className="fadein" key={item.id}>
                  <Grid.Col xs={6} sm={3}>
                    <Icon {...iconProps(item.status)} />
                    {dictionary[item.status] || ''}
                  </Grid.Col>
                  <Grid.Col xs={6} sm={3}>
                    {helpers.getTimeDisplayHowLongAgo(
                      item.status === 'pending' || item.status === 'running' ? item.startTime : item.endTime
                    )}
                  </Grid.Col>
                  <Grid.Col xs={3} sm={2}>
                    <Icon className="scan-job-status-icon systems" type="pf" name="ok" />
                    {item.systemsScanned}
                  </Grid.Col>
                  <Grid.Col xs={3} sm={2}>
                    <Icon className="scan-job-status-icon systems" type="pf" name="error-circle-o" />
                    {item.systemsFailed}
                  </Grid.Col>
                  <Grid.Col xs={3} sm={2}>
                    {item.reportId > 0 && (
                      <ScanDownload downloadName={item.scanName} downloadId={item.reportId} bsStyle="link">
                        <Icon type="pf" name="export" /> &nbsp;Download
                      </ScanDownload>
                    )}
                  </Grid.Col>
                </Grid.Row>
              )
          )}
        </Grid>
      </div>
    );
  }
}

/**
 * Prop types
 *
 * @type {{t: Function, isMoreResults: boolean, pending: boolean, errorMessage: string, getScanJobs: Function,
 *     scanJobsList: Array, id: string|number, error: boolean, mostRecentId: string|number}}
 */
ScanJobsList.propTypes = {
  error: PropTypes.bool,
  errorMessage: PropTypes.string,
  getScanJobs: PropTypes.func,
  isMoreResults: PropTypes.bool,
  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  mostRecentId: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  pending: PropTypes.bool,
  scanJobsList: PropTypes.arrayOf(
    PropTypes.shape({
      endTime: PropTypes.string,
      id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
      reportId: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
      scanName: PropTypes.string,
      startTime: PropTypes.string,
      status: PropTypes.string,
      systemsScanned: PropTypes.number,
      systemsFailed: PropTypes.number
    })
  ),
  t: PropTypes.func
};

/**
 * Default props
 *
 * @type {{t: translate, isMoreResults: boolean, pending: boolean, errorMessage: null, getScanJobs: Function,
 *     scanJobsList: *[], error: boolean, mostRecentId: null}}
 */
ScanJobsList.defaultProps = {
  error: false,
  errorMessage: null,
  getScanJobs: helpers.noop,
  isMoreResults: false,
  mostRecentId: null,
  pending: false,
  scanJobsList: [],
  t: translate
};

const mapDispatchToProps = dispatch => ({
  getScanJobs: (id, query) => dispatch(reduxActions.scans.getScanJobs(id, query))
});

const makeMapStateToProps = () => {
  const getScanJobsDetails = reduxSelectors.scans.makeScanJobsList();

  return (state, props) => ({
    ...getScanJobsDetails(state, props)
  });
};

const ConnectedScanJobsList = connect(makeMapStateToProps, mapDispatchToProps)(ScanJobsList);

export { ConnectedScanJobsList as default, ConnectedScanJobsList, ScanJobsList };
