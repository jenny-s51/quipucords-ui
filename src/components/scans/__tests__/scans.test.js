import React from 'react';
import { Scans } from '../scans';
import { apiTypes } from '../../../constants/apiConstants';

describe('Scans Component', () => {
  it('should render a basic component', async () => {
    const props = {
      useGetScans: () => ({
        fulfilled: true
      })
    };

    const component = await shallowHookComponent(<Scans {...props} />);
    expect(component).toMatchSnapshot('basic');
  });

  it('should handle multiple display states, pending, error, fulfilled', async () => {
    const props = {
      useGetScans: () => ({
        pending: true
      })
    };

    const component = await shallowHookComponent(<Scans {...props} />);
    expect(component).toMatchSnapshot('pending');

    component.setProps({
      useGetScans: () => ({
        pending: false,
        error: true
      })
    });

    expect(component).toMatchSnapshot('error');

    component.setProps({
      useGetScans: () => ({
        pending: false,
        error: false,
        fulfilled: true,
        data: [
          {
            [apiTypes.API_RESPONSE_SCAN_ID]: '1',
            [apiTypes.API_RESPONSE_SCAN_NAME]: 'lorem'
          }
        ]
      })
    });

    expect(component).toMatchSnapshot('fulfilled');
  });

  it('should return an empty state when there are no scans', async () => {
    const props = {
      useGetScans: () => ({
        fulfilled: true,
        data: []
      })
    };

    const component = await shallowHookComponent(<Scans {...props} />);
    expect(component.render()).toMatchSnapshot('empty state');
  });
});
