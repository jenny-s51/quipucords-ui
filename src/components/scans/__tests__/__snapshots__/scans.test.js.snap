// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scans Component should handle multiple display states, pending, error, fulfilled: error 1`] = `
<EmptyState
  className="quipucords-empty-state__alert"
>
  <Alert
    title="t(view.error, {\\"context\\":\\"scans\\"})"
    variant="danger"
  >
    t(view.error-message, {"context":"scans"})
  </Alert>
</EmptyState>
`;

exports[`Scans Component should handle multiple display states, pending, error, fulfilled: fulfilled 1`] = `
<div
  className="quipucords-view-container"
>
  <ViewToolbar
    actions={
      <Tooltip
        content="t(table.tooltip, {\\"context\\":\\"merge-reports\\"})"
        delayShow={500}
        id={null}
        isPopover={false}
        placement="top"
      >
        <Button
          isDisabled={true}
          onClick={[Function]}
          variant="primary"
        >
          t(table.label, {"context":"merge-reports"})
        </Button>
      </Tooltip>
    }
    activeFilters={Array []}
    filterFields={
      Array [
        Object {
          "filterType": "text",
          "id": "search_by_name",
          "placeholder": "Filter by Name",
          "title": "Name",
        },
        Object {
          "filterType": "text",
          "id": "search_sources_by_name",
          "placeholder": "Filter by Source Name",
          "title": "Source",
        },
      ]
    }
    filterType={Object {}}
    filterValue=""
    itemsType="Scan"
    itemsTypePlural="Scans"
    lastRefresh={NaN}
    onRefresh={[Function]}
    selectedCount={0}
    sortAscending={true}
    sortFields={
      Array [
        Object {
          "id": "name",
          "isNumeric": false,
          "title": "Name",
        },
        Object {
          "id": "most_recent_scanjob__start_time",
          "isNumeric": true,
          "sortAscending": false,
          "title": "Most Recent",
        },
      ]
    }
    sortType={Object {}}
    totalCount={0}
    viewType="SCANS_VIEW"
  />
  <ViewPaginationRow
    currentPage={0}
    pageSize={0}
    totalCount={0}
    viewType="SCANS_VIEW"
  />
  <div
    className="quipucords-list-container"
  >
    <Table
      ariaLabel={null}
      className=""
      columnHeaders={Array []}
      componentClassNames={
        Object {
          "table": "quipucords-table",
          "td": "quipucords-table__td",
          "tdAction": "quipucords-table__td-action",
          "tdExpand": "quipucords-table__td-expand",
          "tdExpanded": "quipucords-table__td-expand-expanded",
          "tdExpandedContent": "quipucords-table__td-expand-content",
          "tdExpandedWrapper": "quipucords-table__td-expand-wrapper",
          "tdSelect": "quipucords-table__td-select",
          "th": "quipucords-table__th",
          "tr": "quipucords-table__tr",
          "trExpand": "quipucords-table__tr-expand",
          "trExpanded": "quipucords-table__tr-expand-expanded",
          "trExpandedContent": "quipucords-table__tr-expand-content",
        }
      }
      isBorders={true}
      isHeader={false}
      onExpand={[Function]}
      onSelect={[Function]}
      onSort={null}
      rows={
        Array [
          Object {
            "cells": Array [
              Object {
                "content": <Grid
                  hasGutter={false}
                >
                  <GridItem
                    sm={2}
                  />
                  <GridItem
                    sm={10}
                  >
                    <div>
                      <strong>
                        lorem
                      </strong>
                    </div>
                  </GridItem>
                </Grid>,
                "dataLabel": "t(table.header, {\\"context\\":\\"description\\"})",
              },
              Object {
                "content": <Grid
                  hasGutter={false}
                >
                  <GridItem
                    sm={2}
                  >
                    <ContextIcon
                      size="sm"
                      symbol={null}
                    />
                  </GridItem>
                  <GridItem
                    sm={10}
                  >
                    <div>
                      t(table.label_status, {"context":"scans"})
                    </div>
                    a day ago
                  </GridItem>
                </Grid>,
                "dataLabel": "t(table.header, {\\"context\\":\\"scan\\"})",
                "width": 20,
              },
              Object {
                "content": <Tooltip
                  content="t(table.label_status_tooltip_success, {\\"context\\":\\"scans\\",\\"count\\":0})"
                  delayShow={500}
                  id={null}
                  isPopover={false}
                  placement="top"
                >
                  t(table.label_status_cell, {"context":"scans","count":0}, [object Object],[object Object])
                </Tooltip>,
                "dataLabel": "t(table.header_success, {\\"context\\":\\"scans\\"})",
                "expandedContent": undefined,
                "isExpanded": false,
                "width": 8,
              },
              Object {
                "content": <Tooltip
                  content="t(table.label_status_tooltip_failed, {\\"context\\":\\"scans\\",\\"count\\":0})"
                  delayShow={500}
                  id={null}
                  isPopover={false}
                  placement="top"
                >
                  t(table.label_status_cell, {"context":"scans","count":0}, [object Object],[object Object])
                </Tooltip>,
                "dataLabel": "t(table.header_failed, {\\"context\\":\\"scans\\"})",
                "expandedContent": undefined,
                "isExpanded": false,
                "width": 8,
              },
              Object {
                "content": <Tooltip
                  content="t(table.label_status_tooltip_sources, {\\"context\\":\\"scans\\",\\"count\\":0})"
                  delayShow={500}
                  id={null}
                  isPopover={false}
                  placement="top"
                >
                  t(table.label_status_cell, {"context":"scans","count":0}, [object Object],[object Object])
                </Tooltip>,
                "dataLabel": "t(table.header, {\\"context\\":\\"sources\\"})",
                "expandedContent": undefined,
                "isExpanded": false,
                "width": 8,
              },
              Object {
                "content": <Tooltip
                  content="t(table.label_status_tooltip_scans, {\\"context\\":\\"scans\\",\\"count\\":0})"
                  delayShow={500}
                  id={null}
                  isPopover={false}
                  placement="top"
                >
                  t(table.label_status_cell, {"context":"scans","count":0}, [object Object],[object Object])
                </Tooltip>,
                "dataLabel": "t(table.header, {\\"context\\":\\"scan-jobs\\"})",
                "expandedContent": undefined,
                "isExpanded": false,
                "width": 8,
              },
              Object {
                "content": <OverflowMenu
                  breakpoint="lg"
                >
                  <OverflowMenuContent>
                    <OverflowMenuGroup
                      groupType="button"
                    />
                  </OverflowMenuContent>
                  <OverflowMenuControl>
                    <DropdownSelect
                      ariaLabel="Select option"
                      buttonVariant="plain"
                      className=""
                      direction="down"
                      id="generatedid-"
                      isDisabled={false}
                      isDropdownButton={true}
                      isInline={true}
                      isToggleText={true}
                      maxHeight={null}
                      name={null}
                      onSelect={[Function]}
                      options={Array []}
                      placeholder={
                        <EllipsisVIcon
                          color="currentColor"
                          noVerticalAlign={false}
                          size="sm"
                        />
                      }
                      position="right"
                      selectedOptions={null}
                      splitButtonVariant={null}
                      toggleIcon={null}
                      variant="single"
                    />
                  </OverflowMenuControl>
                </OverflowMenu>,
                "isActionCell": true,
                "style": Object {
                  "textAlign": "right",
                },
              },
            ],
            "isSelected": false,
            "item": Object {
              "id": "1",
              "name": "lorem",
            },
          },
        ]
      }
      summary={null}
      variant="compact"
    >
      <ScansEmptyState
        getScansSources={[Function]}
        history={Object {}}
        sourcesCount={0}
        sourcesExist={false}
        t={[Function]}
        uiShortName="Quipucords"
        viewId="scans"
      />
    </Table>
  </div>
</div>
`;

exports[`Scans Component should handle multiple display states, pending, error, fulfilled: pending 1`] = `
<Modal
  aria-label={null}
  backdrop={false}
  className={null}
  disableFocusTrap={true}
  footer={null}
  header={null}
  isContentOnly={false}
  isOpen={true}
  position="top"
  positionOffset="5%"
  showClose={false}
  t={[Function]}
  variant="medium"
>
  <Spinner
    isSVG={true}
    size="lg"
  />
  <div
    className="text-center"
  >
    t(view.loading, {"context":"scans"})
  </div>
</Modal>
`;

exports[`Scans Component should render a basic component: basic 1`] = `
<div
  className="quipucords-view-container"
>
  <div
    className="quipucords-list-container"
  >
    <Table
      ariaLabel={null}
      className=""
      columnHeaders={Array []}
      componentClassNames={
        Object {
          "table": "quipucords-table",
          "td": "quipucords-table__td",
          "tdAction": "quipucords-table__td-action",
          "tdExpand": "quipucords-table__td-expand",
          "tdExpanded": "quipucords-table__td-expand-expanded",
          "tdExpandedContent": "quipucords-table__td-expand-content",
          "tdExpandedWrapper": "quipucords-table__td-expand-wrapper",
          "tdSelect": "quipucords-table__td-select",
          "th": "quipucords-table__th",
          "tr": "quipucords-table__tr",
          "trExpand": "quipucords-table__tr-expand",
          "trExpanded": "quipucords-table__tr-expand-expanded",
          "trExpandedContent": "quipucords-table__tr-expand-content",
        }
      }
      isBorders={true}
      isHeader={false}
      onExpand={[Function]}
      onSelect={[Function]}
      onSort={null}
      rows={Array []}
      summary={null}
      variant="compact"
    >
      <ScansEmptyState
        getScansSources={[Function]}
        history={Object {}}
        sourcesCount={0}
        sourcesExist={false}
        t={[Function]}
        uiShortName="Quipucords"
        viewId="scans"
      />
    </Table>
  </div>
</div>
`;

exports[`Scans Component should return an empty state when there are no scans: empty state 1`] = `
<div
  class="quipucords-view-container"
>
  <div
    class="quipucords-list-container"
  >
    <div
      class="pf-l-grid"
    >
      <div
        class="pf-l-grid__item pf-m-12-col"
      >
        <div
          class="pf-c-empty-state pf-m-lg quipucords-empty-state"
        >
          <div
            class="pf-c-empty-state__content"
          >
            <svg
              aria-hidden="true"
              class="pf-c-empty-state__icon"
              fill="currentColor"
              height="1em"
              role="img"
              style="vertical-align:-0.125em"
              viewBox="0 0 1024 1024"
              width="1em"
            >
              <path
                d="M576,303 C576,294.715729 569.284271,288 561,288 L463,288 C454.715729,288 448,294.715729 448,303 L448,448 L303,448 C294.715729,448 288,454.715729 288,463 L288,561 C288,569.284271 294.715729,576 303,576 L448,576 L448,720.9 C447.983373,729.207373 454.6927,735.961429 463,736 L561,736 C569.3073,735.961429 576.016627,729.207373 576,720.9 L576,576 L721,576 C724.969024,576.026638 728.784638,574.468589 731.600595,571.671405 C734.416553,568.87422 736.000031,565.069113 736.000031,561.1 L736.000031,463.1 C736.016627,454.792627 729.3073,448.038571 721,448 L576,448 L576,303 Z M512,896 C300.2,896 128,723.9 128,512 C128,300.3 300.2,128 512,128 C723.8,128 896,300.2 896,512 C896,723.8 723.7,896 512,896 Z M512.1,0 C229.7,0 0,229.8 0,512 C0,794.2 229.8,1024 512.1,1024 C794.4,1024 1024,794.3 1024,512 C1024,229.7 794.4,0 512.1,0 Z"
              />
            </svg>
            <h1
              class="pf-c-title pf-m-2xl"
              data-ouia-component-id="OUIA-Generated-Title-1"
              data-ouia-component-type="PF4/Title"
              data-ouia-safe="true"
            >
              t(view.empty-state_title, {"context":"scans","count":0,"name":"Quipucords"})
            </h1>
            <div
              class="pf-c-empty-state__body"
            >
              t(view.empty-state_description, {"context":"scans","count":0})
            </div>
            <div
              class="pf-c-empty-state__primary"
            >
              <button
                aria-disabled="false"
                class="pf-c-button pf-m-primary"
                data-ouia-component-id="OUIA-Generated-Button-primary-1"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                type="button"
              >
                t(view.empty-state_label, {"context":"source-navigate","count":0})
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
